narwhals/dataframe.py

```diff
@@ -498,11 +498,13 @@ def __arrow_c_stream__(self: Self, requested_schema: object | None = None) -> ob
         pa_table = self.to_arrow()
         return pa_table.__arrow_c_stream__(requested_schema=requested_schema)
 
-    def lazy(self: Self) -> LazyFrame[Any]:
+    def lazy(self: Self, *, backend: Implementation | None = None) -> LazyFrame[Any]:
         """Lazify the DataFrame (if possible).
 
         If a library does not support lazy execution, then this is a no-op.
 
+        If `backend` is specified some conversions between libraries can occur.
+
```

Comment:
What about something along the following lines:

```suggestion
        If `backend` is specified, then a conversion between different backends
        might be triggered.
        If a library does not support lazy execution and `backend` is not specified,
        then this is a no-op.
        
        Arguments:
            backend: specifies which Implementation to convert to.
```

Feel free to expand 😇 (also I thought we have a linter to check for documentation of Arguments 🤔